sources/Renderer/Vulkan/VKSwapChain.cpp

```diff
@@ -370,16 +374,34 @@ void VKSwapChain::CreateGpuSurface()
 
     #elif defined LLGL_OS_LINUX
 
-    VkXlibSurfaceCreateInfoKHR createInfo;
+    // Create Xlib surface
+    if (nativeHandle.type == NativeHandleType::X11)
     {
+        VkXlibSurfaceCreateInfoKHR createInfo;
         createInfo.sType    = VK_STRUCTURE_TYPE_XLIB_SURFACE_CREATE_INFO_KHR;
         createInfo.pNext    = nullptr;
         createInfo.flags    = 0;
-        createInfo.dpy      = nativeHandle.display;
-        createInfo.window   = nativeHandle.window;
+        createInfo.dpy      = nativeHandle.x11.display;
+        createInfo.window   = nativeHandle.x11.window;
+
+        VkResult result = vkCreateXlibSurfaceKHR(instance_, &createInfo, nullptr, surface_.ReleaseAndGetAddressOf());
+        VKThrowIfFailed(result, "failed to create Xlib surface for Vulkan swap-chain");
     }
-    VkResult result = vkCreateXlibSurfaceKHR(instance_, &createInfo, nullptr, surface_.ReleaseAndGetAddressOf());
-    VKThrowIfFailed(result, "failed to create Xlib surface for Vulkan swap-chain");
+    #ifdef LLGL_LINUX_ENABLE_WAYLAND
+    // Create Wayland surface
+    else if (nativeHandle.type == NativeHandleType::Wayland)
```

Comment:
Since we currently always support X11, can we only check for Wayland and use X11 in the else branch or at least have an else branch in case `nativeHandle` is corrupted. That else branch can simply contain `LLGL_TRAP("Invalid native handle type");` or something like that.